agent/command/git.go

```diff
@@ -870,15 +871,34 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		}
 	}
 
-	// Clone the project's modules.
+	// For every module, expand the module prefix.
 	for _, moduleName := range conf.BuildVariant.Modules {
-		if err := ctx.Err(); err != nil {
-			return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
-		}
-		err = c.fetchModuleSource(ctx, comm, conf, logger, jpm, td, opts.token, opts.method, p, moduleName)
+		module, err := conf.Project.GetModuleByName(moduleName)
 		if err != nil {
-			return errors.Wrapf(err, "fetching module source '%s'", moduleName)
+			return errors.Wrapf(err, "getting module '%s'", moduleName)
+		}
+		if module == nil {
+			return errors.Errorf("module '%s' not found", moduleName)
 		}
+		expandModulePrefix(conf, module.Name, module.Prefix, logger)
+	}
+
+	g, gCtx := errgroup.WithContext(ctx)
+	g.SetLimit(10)
+
+	// Clone the project's modules in goroutines.
+	for _, mN := range conf.BuildVariant.Modules {
```

Comment:
nit: mN is kind of a non-standard variable name, could we just use "module" or "name" instead? Also, does this need to be reassigned to a variable in 891 so that it isn't over-written for use inside the go routine? could we add a comment to that effect, since it's unusual to do for non-pointers?

Reply:
Sure, I'll also link our golang upgrade ticket since 1.22 golang resolves this exact issue and it won't be needed anymore in the comment